File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.py
def solution(number, limit, power):
    answer = 0
    result=[]
    for num in range(1,number+1):
        dirtn=0
        for i in range (1,int(num**0.5)+1):
            if num%i==0: dirtn+=1
        if int(num**0.5)==num**0.5:
            if dirtn*2-1>limit:
                answer+=power
            else:answer+=dirtn*2-1
            
        else : 
            if dirtn*2>limit:
                answer+=power
            else:
                answer+=dirtn*2
        
            
    return answer

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.java
class Solution {
    public int solution(int number, int limit, int power) {
        int answer = 0;
    
        for(int kisa=1;kisa<=number;kisa++){
            int num=0; //약수개수
            for(int p=1;p<=Math.sqrt(kisa);p++){
                if(kisa%p==0){
                    num+=2;
                }
            }
            if(Math.sqrt(kisa)==(int)Math.sqrt(kisa)) num-=1;
            
            if(num>limit) answer+=power;
            else answer+=num;
        }
        return answer;
    }
}